fixups/FileRedirectionFixup/PathRedirection.cpp

```diff
@@ -349,6 +339,62 @@ std::wstring RedirectedPath(const normalized_path& deVirtualizedPath, bool ensur
     return result;
 }
 
+/// <summary>
+/// Figures out the absolute path to redirect to.
+/// </summary>
+/// <param name="deVirtualizedPath">The original path from the app</param>
+/// <param name="ensureDirectoryStructure">If true, the deVirtualizedPath will be appended to the allowed write location</param>
+/// <returns>The new absolute path.</returns>
+std::wstring RedirectedPath(const normalized_path& deVirtualizedPath, bool ensureDirectoryStructure)
+{
+    /*
+        To prevent apps breaking on an upgrade we redirect writes to the package root path to
+        a path that contains the package family name and not the package full name.
+    */
+    std::wstring result;
+    bool shouldredirectToPackageRoot = false;
+    if (deVirtualizedPath.full_path.find(g_packageRootPath) != std::wstring::npos)

```

Comment:
Let's create a backlog item that aims to measure the perf impact on typical apps for this find() call.